ibis/backends/pandas/executor/core.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from functools import singledispatch
+
+import ibis.expr.operations as ops
+from ibis.backends.pandas.rewrites import plan
+from ibis.common.exceptions import OperationNotDefinedError
+from ibis.formats.pandas import PandasData
+
+
+@singledispatch
+def execute(node, **kwargs):
+    raise OperationNotDefinedError(f"no rule for {type(node)}")
+
+
+def zuper(node, params):
```

Comment:
Heh, this looks a lot like the SQL backend translation :)

Reply:
I moved this function to `Executor.execute()`